pkg/maps/ctmap/ctmap.go

```diff
@@ -250,16 +246,17 @@ func doGC6(m *bpf.Map, filter *GCFilter) int {
 			del = true
 		}
 
-		n := nextKey.ToHost().(*CtKey6Global)
 		if filter.fType&GCFilterByID != 0 &&
 			// In CT's entries, saddr is the packet's receiver,
 			// which means, is the destination container IP.
-			n.saddr.IP().Equal(filter.IP) {
+			nextKey.saddr.IP().Equal(filter.IP) {
 
 			ruleCtx := policy.RuleContext{
-				SecID: policy.NumericIdentity(entry.src_sec_id),
-				Port:  n.sport,
-				Proto: uint8(n.nexthdr),
+				SecID:          policy.NumericIdentity(entry.src_sec_id),
+				Port:           nextKey.sport,
+				Proto:          uint8(nextKey.nexthdr),
+				L7RedirectPort: entry.proxy_port,
```

Comment:
I assume that `CtEntry.proxy_port` is in network byte-order also? We should double-check all of the usages of it, and also consider updating the struct definition to have a comment describing the byte-order of this field.